app/pages/sections/items/ItemExciseProductCodePage.scala

```diff
@@ -16,11 +16,39 @@
 
 package pages.sections.items
 
-import models.Index
+import config.Constants.BODYEADESAD
+import models.requests.DataRequest
+import models.{Index, MovementSubmissionFailure}
 import pages.QuestionPage
 import play.api.libs.json.JsPath
+import utils.SubmissionFailureErrorCodes._
 
+// indexesOfMovementSubmissionErrors is not needed here because when the user
+// changes their answer, all the errors for this item are deleted
 case class ItemExciseProductCodePage(idx: Index) extends QuestionPage[String] {
   override val toString: String = "itemExciseProductCode"
   override val path: JsPath = ItemsSectionItem(idx).path \ toString
+
+  private val itemExciseProductCodesError: Seq[String] = Seq(
+    itemExciseProductCodeConsignorNotApprovedToSendError,
+    itemExciseProductCodeConsigneeNotApprovedToReceiveError,
+    itemExciseProductCodeDestinationNotApprovedToReceiveError,
+    itemExciseProductCodeDispatchPlaceNotAllowed
+  )
+
+  private def isExciseProductCodeAtIndex: MovementSubmissionFailure => Boolean = submissionFailure =>
+    submissionFailure.errorLocation.exists(_.contains(s"$BODYEADESAD[${idx.position + 1}]")) &&
```

Comment:
Not sure if it's over-abstraction, but we've got 5 places in the repo that do this kind of check. Maybe a little helper method somewhere? Also wondering if the BODYEADESAD string could be a constant, but see we've already got a Pascal case one in there. Would it be worth adding a constant and differentiating between the two?

Reply:
Done in 3cada8db283b4cf563861e068084de24e44a5b24